pkg/kubelet/config/common.go

```diff
@@ -144,19 +149,11 @@ func tryDecodeSinglePod(data []byte, defaultFn defaultFunc) (parsed bool, pod *v
 		return true, nil, err
 	}
 
-	for _, v := range v1Pod.Spec.Volumes {
```

Comment:
this code is already unconditional, we don't want to drop these checks if the PreventStaticPodAPIReferences gate is toggled off, leave this in an `else` block with a comment to remove once PreventStaticPodAPIReferences is GA and locked on